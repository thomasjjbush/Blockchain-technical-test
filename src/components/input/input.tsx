import React, { ChangeEvent, FC, ReactElement } from 'react';
import styled from 'styled-components';
import { SearchInputProps as Props } from '../../../types';

const Container = styled.span`
    border: solid 2px ${(props): string => props.theme.colors.offset};
    border-radius: 5px;
    overflow: hidden;
`;

const Button = styled.button`
    border: none;
    height: 100%;
    padding: 0 10px;
`;

const Input = styled.input`
    border: none;
    height: 100%;
    padding: 0 10px;
`;

export const SearchInput: FC<Props> = ({ onChange, value, placeholder }: Props): ReactElement => {
    return (
        <Container>
            <Input
                data-test-id="search"
                onChange={(e: ChangeEvent<HTMLInputElement>) => onChange(e.currentTarget.value)}
                placeholder={placeholder}
                value={value}
                type="text"
            />
            <Button data-test-id="clear" disabled={!value} onClick={() => onChange('')}>
                x
            </Button>
        </Container>
    );
};
