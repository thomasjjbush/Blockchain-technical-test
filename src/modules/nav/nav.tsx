import React, { FC, ReactElement, useState } from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import { SearchInput } from '../../components/input/input';

const Button = styled(Link)`
    background-color: ${(props): string => props.theme.colors.brand};
    border-radius: 5px;
    color: white;
    line-height: 30px;
    padding: 0 15px;
    margin-left: 10px;
`;

const Container = styled.nav`
    background-color: white;
    box-shadow: 5px 0px 10px 0px rgba(0, 0, 0, 0.3);
    display: flex;
    height: 50px;
    justify-content: center;
    padding: 10px;
`;

export const Nav: FC = (): ReactElement => {
    const [address, setAddress] = useState('');

    return (
        <Container>
            <SearchInput onChange={setAddress} placeholder="Enter Bitcoin address" value={address} />
            <Button data-test-id="go" to={`/address/${address}`}>
                Go!
            </Button>
        </Container>
    );
};
