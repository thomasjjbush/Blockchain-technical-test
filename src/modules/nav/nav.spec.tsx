import React from 'react';
import { shallow } from 'enzyme';
import { Nav } from './nav';
import { SearchInput } from '../../components/input/input';

describe('Nav', () => {
    it('should match snapshot', () => {
        expect(shallow(<Nav />)).toMatchSnapshot();
    });

    it('should update address on input change', () => {
        const wrapper = shallow(<Nav />);
        expect(wrapper.find({ 'data-test-id': 'go' }).prop('to')).toBe('/address/');

        wrapper.find(SearchInput).prop('onChange')('888');
        expect(wrapper.find({ 'data-test-id': 'go' }).prop('to')).toBe('/address/888');
    });
});
