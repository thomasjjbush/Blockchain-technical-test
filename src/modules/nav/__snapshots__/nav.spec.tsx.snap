// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match snapshot 1`] = `
<styled.nav
  styles="
    background-color: white;
    box-shadow: 5px 0px 10px 0px rgba(0, 0, 0, 0.3);
    display: flex;
    height: 50px;
    justify-content: center;
    padding: 10px;
"
>
  <SearchInput
    onChange={[Function]}
    placeholder="Enter Bitcoin address"
    value=""
  />
  <Styled(Link)
    data-test-id="go"
    styles="
    background-color: theme.colors.brand;
    border-radius: 5px;
    color: white;
    line-height: 30px;
    padding: 0 15px;
    margin-left: 10px;
"
    to="/address/"
  >
    Go!
  </Styled(Link)>
</styled.nav>
`;
